src/Components/CopyToClipboard.jsx

```diff
@@ -1,13 +1,36 @@
+import { useState, useEffect } from "react";
+
 export default function CopyToClipboard({ copiedText }) {
+  const [isCopy, setIsCopy] = useState(false);
   /* handle the click event for copy button */
   const handleCopyClick = async () => {
     try {
       await navigator.clipboard.writeText(copiedText);
+
+      setIsCopy(true);
     } catch (error) {
-      console.error("Copy failed");
+      console.error("Copy failed", error);
     }
   };
-  handleCopyClick();
 
-  return <button onClick={handleCopyClick}>COPY</button>;
+  useEffect(() => {
+    if (isCopy) {
+      const timer = setTimeout(() => {
+        setIsCopy(false);
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
```

Comment:
nice! There is something that I personally really like in code and that is keeping your code as "flat" as possible, since it improves readability, in this case, we can reduce the level of nesting by one but switching the logic around: 
```js
if(!isCopy) {
  return 
}

const timer = setTimeout(() => {
    setIsCopy(false);
}, 3000);

return () => clearTimeout(timer);

```